### PR TITLE
New theme selection UI

### PR DESCRIPTION
**Pull request by chingucoding**
This PR replaces the existing theme choose UI with a theme sample preview grid:

Screenshot of new theme selection view in settings

## Open issues
* [ ] AccentColor preview behaves weird, when the selected theme overrides it, it might be picked up by the samples
* [x] The selection indicator of the GridView is too subtle
* [x] The BlackPixel theme is behaving strange, maybe we need to take a look at that theme again and fix it?
* [ ] Is the preview UI good? Could it be improved?

### CODE REVIEW

**Comment by yaichenbaum** [comment]
> The BlackPixel theme is behaving strange, maybe we need to take a look at that theme again and fix it?

The black pixel theme is outdated and will likely have to be removed.

**Comment by XPoppyX** [comment]
> * Is the preview UI good? Could it be improved?

Maybe rounded corners and border? if possible

Captura de pantalla 2021-07-19 185659 Captura de pantalla 2021-07-19 190254

**Comment by yaichenbaum** [comment]
@XPoppyX the preview has rounded corners on Windows 11.

**Comment by chingucoding** [comment]
> Maybe rounded corners and border? if possible

Very good idea @XPoppyX , add this now.

Also fixed some issues.

File: Files/App.xaml [hunk 1]
```diff
@@ -166,6 +166,7 @@
                             <Color x:Key="LayerFillColorDefault">#11000000</Color>
                         </ResourceDictionary>
                     </ResourceDictionary.ThemeDictionaries>
+                    <x:Double x:Key="SettingsBlockControlDefaultWidth">385</x:Double>
```
Comment (yaichenbaum) [inline Files/App.xaml:169]: Should this be 396?

Comment (chingucoding) [reply Files/App.xaml:169]: I just took a value that fitted, there was no deeper though process than "Yup this works".

Comment (yaichenbaum) [reply Files/App.xaml:169]: We try to make sure most values are divisible by 4 so I think we should try to match that.

Comment (chingucoding) [reply Files/App.xaml]: Alrighty updated the values.

File: Files/UserControls/Settings/ThemeSampleDisplayControl.xaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            <RowDefinition Height="Auto"/>
+        </Grid.RowDefinitions>
+        <Grid CornerRadius="4" BorderThickness="1" 
+            BorderBrush="{ThemeResource SystemControlBackgroundBaseMediumHighBrush}">
+            <Grid.RowDefinitions>
+                <RowDefinition Height="20px"/>
+                <RowDefinition Height="*"/>
+            </Grid.RowDefinitions>
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="35px"/>
```
Comment (yaichenbaum) [inline Files/UserControls/Settings/ThemeSampleDisplayControl.xaml:24]:
```suggestion
                <ColumnDefinition Width="36px"/>
```

File: Files/UserControls/Settings/ThemeSampleDisplayControl.xaml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            <Grid x:Name="LeftSidePanel" Grid.ColumnSpan="2" Grid.Row="1" Background="{ThemeResource PaneHolderPageBackgroundBrush}">
+                <Grid.RowDefinitions>
+                    <RowDefinition Height="10"/>
+                    <RowDefinition Height="10"/>
+                    <RowDefinition Height="*"/>
+                    <RowDefinition />
+                </Grid.RowDefinitions>
+
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition Width="3"/>
```
Comment (yaichenbaum) [inline Files/UserControls/Settings/ThemeSampleDisplayControl.xaml:37]:
```suggestion
                    <ColumnDefinition Width="4"/>
```

File: Files/UserControls/Settings/ThemeSampleDisplayControl.xaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                    Fill="{ThemeResource SolidBackgroundFillColorSecondaryBrush}"/>
+                <Rectangle HorizontalAlignment="Stretch" VerticalAlignment="Stretch" Grid.RowSpan="4" Grid.ColumnSpan="4"
+                    Fill="{ThemeResource HorizontalTabControlBackgroundBrush}"/>
+                <Rectangle HorizontalAlignment="Stretch" VerticalAlignment="Stretch" Grid.RowSpan="4" Grid.ColumnSpan="4"
+                    Fill="{ThemeResource PaneHolderPageBackgroundBrush}"/>
+
+                <Rectangle Grid.Row="0" Grid.Column="1"
+                    Fill="{ThemeResource SystemAccentColor}" RadiusX="2" RadiusY="2" Height="6" Width="6"/>
+                <Rectangle Grid.Row="0" Grid.Column="2"
+                    Fill="{ThemeResource SystemControlForegroundBaseMediumBrush}" RadiusX="1" RadiusY="1" Height="2" Width="15"/>
```
Comment (yaichenbaum) [inline Files/UserControls/Settings/ThemeSampleDisplayControl.xaml:53]:
```suggestion
                    Fill="{ThemeResource SystemControlForegroundBaseMediumBrush}" RadiusX="1" RadiusY="1" Height="2" Width="16"/>
```

File: Files/UserControls/Settings/ThemeSampleDisplayControl.xaml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+                <Rectangle Grid.Row="1" Grid.Column="1"
+                    Fill="{ThemeResource SystemAccentColor}" RadiusX="2" RadiusY="2" Height="6" Width="6"/>
+                <Rectangle Grid.Row="1" Grid.Column="2"
+                    Fill="{ThemeResource SystemControlForegroundBaseMediumBrush}" RadiusX="1" RadiusY="1" Height="2" Width="15"/>
```
Comment (yaichenbaum) [inline Files/UserControls/Settings/ThemeSampleDisplayControl.xaml:58]:
```suggestion
                    Fill="{ThemeResource SystemControlForegroundBaseMediumBrush}" RadiusX="1" RadiusY="1" Height="2" Width="16"/>
```

File: Files/UserControls/Settings/ThemeSampleDisplayControl.xaml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+                <Rectangle Grid.Row="1" Grid.Column="1"
+                    Fill="{ThemeResource SystemAccentColor}" RadiusX="2" RadiusY="2" Height="6" Width="6"/>
+                <Rectangle Grid.Row="1" Grid.Column="2"
+                    Fill="{ThemeResource SystemControlForegroundBaseMediumBrush}" RadiusX="1" RadiusY="1" Height="2" Width="16"/>
+                
+                <Rectangle Grid.Row="3" Grid.Column="1"
+                    Fill="{ThemeResource SystemControlForegroundBaseMediumBrush}" RadiusX="2" RadiusY="2" Height="6" Width="6"/>
+                <Rectangle Grid.Row="3" Grid.Column="2"
+                    Fill="{ThemeResource SystemControlForegroundBaseMediumBrush}" RadiusX="1" RadiusY="1" Height="2" Width="15"/>
```
Comment (yaichenbaum) [inline Files/UserControls/Settings/ThemeSampleDisplayControl.xaml:63]:
```suggestion
                    Fill="{ThemeResource SystemControlForegroundBaseMediumBrush}" RadiusX="1" RadiusY="1" Height="2" Width="16"/>
```

**Comment by yaichenbaum** [comment]
@chingucoding can you update the setting cards on all the setting pages to match the new width?

**Comment by chingucoding** [comment]
> @chingucoding can you update the setting cards on all the setting pages to match the new width?

Done!

**Review by yaichenbaum — APPROVED** [review]
Great work! LGTM

**Pull request merged**